e2e/cypress/integration/messaging/emoji_keyboard_entry_spec.js

```diff
@@ -0,0 +1,133 @@
+
+describe('Message', () => {
+    beforeEach(() => {
+        // # Login as "user-1" and go to /
+        cy.apiLogin('user-1');
+        cy.visit('/');
+
+        // # Open emoji picker
+        cy.get('#emojiPickerButton').click();
+
+        // # Wait for emojis to load
+        cy.get('#emojiPicker').should('be.visible');
+
+        // # Select "People" category
+        cy.get('#emojiPickerCategories').find('.fa.fa-smile-o').click();
+    });
+
+    describe('starting with no prior selection', () => {
+        ['right', 'down'].forEach((dir) => {
+            it(`${dir} key press should select the first emoji`, () => {
+                // # Type directional arrow key
+                cy.get('#emojiPickerSearch').type(`{${dir}arrow}`);
+
+                // * The the first emoji in the "People" category should be selected
+                cy.get('.emoji-picker__item.selected').then((selectedEmoji) => {
+                    expect(selectedEmoji.index()).to.equal(0);
+                });
+            });
+        });
+
+        ['left', 'up'].forEach((dir) => {
+            it(`${dir} key press should select nothing`, () => {
+                // # Type directional arrow key
+                cy.get('#emojiPickerSearch').type(`{${dir}arrow}`);
+
+                // * There should be no selection
+                cy.get('.emoji-picker__item.selected').should('not.exist');
+            });
+        });
+    });
+
+    describe('starting with first emoji selected', () => {
+        beforeEach(() => {
+            // # Type right arrow key
+            cy.get('#emojiPickerSearch').type('{rightarrow}');
+        });
+
+        it('right key press should select the second emoji in the list', () => {
+            // # Type directional arrow key
+            cy.get('#emojiPickerSearch').type('{rightarrow}');
+
+            // * Second emoji should now be selected
+            cy.get('.emoji-picker__item.selected').then((selectedEmoji) => {
+                expect(selectedEmoji.index()).to.equal(1);
+            });
+        });
+
+        it('down key press should select emoji >= 6 elements away in list', () => {
+            // # Type directional arrow key
+            cy.get('#emojiPickerSearch').type('{downarrow}');
+
+            // * Down arrow should select next row, index distance is dependant on browser size
+            cy.get('.emoji-picker__item.selected').then((selectedEmoji) => {
+                expect(selectedEmoji.index()).to.greaterThan(5);
+            });
+        });
+
+        ['left', 'up'].forEach((dir) => {
+            it(`selection should remain at index 0 on ${dir} key press`, () => {
+                // # Type directional arrow key
+                cy.get('#emojiPickerSearch').type(`{${dir}arrow}`);
+
+                // * Index should still equal 0
+                cy.get('.emoji-picker__item.selected').then((selectedEmoji) => {
+                    expect(selectedEmoji.index()).to.equal(0);
```

Comment:
The 5 lines above seem like they can be pulled into a single function for simplicity since they're being used in multiple places with different parameters.

Reply:
Makes sense.  I'll get that cleaned up when I get a chance.